keyboards/handwired/mingwahlu/mingwahlumacro/keymaps/default/keymap.c

```diff
@@ -0,0 +1,184 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL,
+    M_NUTS,
+    M_RESET,
+    M_CLOSETAB,
+    M_REOPENTAB,
+    M_NEWTAB,
+    M_SWITCHTABS,
+    M_ALTTAB,
+    M_ALTF4,
+    M_UMARUBM,
+    M_UMARUEAT,
+    M_UMARUMAD,
+    M_UMARUHAPPY,
+    M_UMARUSAD,
+    M_UMARUGAME,
+    M_UMARUSLEEP
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        M_UMARUHAPPY, M_UMARUSAD, M_UMARUMAD, MO(_FN),
+        M_UMARUEAT, M_UMARUGAME, M_UMARUBM, M_UMARUSLEEP
+    ),
+    [_FN] = LAYOUT(
+        M_ALTF4, KC_F5, M_NUTS, MO(_FN),
+        M_CLOSETAB, M_REOPENTAB, M_NEWTAB, M_SWITCHTABS
+    )
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case QMKBEST:
+            if (record->event.pressed) {
+                // when keycode QMKBEST is pressed
+                SEND_STRING("QMK is the best thing ever!");
+            } else {
+                // when keycode QMKBEST is released
+            }
+            break;
+        case QMKURL:
+            if (record->event.pressed) {
+                // when keycode QMKURL is pressed
+                SEND_STRING("https://qmk.fm/\n");
+            } else {
+                // when keycode QMKURL is released
+            }
+            break;
+        case M_NUTS:
+            if (record->event.pressed) {
+                SEND_STRING("deeznuts");
+                tap_code(KC_ENTER);
+            } else {
+            }
+            break;
+        case M_RESET:
+            if (record->event.pressed) {
+                reset_keyboard();
+            } else {
+            }
+            break;
+        case M_CLOSETAB:
+            if(record->event.pressed) {
+                SEND_STRING(SS_LCTL("w"));
+            } else {
+
+            }
+            break;
+        case M_REOPENTAB:
+            if(record->event.pressed) {
+                SEND_STRING(SS_LCTL(SS_LSFT("t")));
+            } else {
+
+            }
+            break;
+        case M_NEWTAB:
+            if(record->event.pressed) {
+                SEND_STRING(SS_LCTL("t"));
```

Comment:
This can be replaced with `LCTL(KC_T)` in the keymap.

Reply:
macros have been removed, and we are now using LCTL(KC_T) in the keymap.